example/ram/compile-tests.lurk

```diff
@@ -0,0 +1,172 @@
+!(:load "quasi.lurk")
+!(:load "compile.lurk")
+
+!(:assert-eq
+  1
+  (compile-closure
+   1
+   '(b)
+   '()
+   '((a . 1))))
+
+!(:assert-eq
+  1
+  (compile-closure
+   'a
+   '(b)
+   '()
+   '((a . 1))))
+
+!(:assert-eq
+  '(+ 1 b)
+  (compile-closure
+   '(+ a b)
+   '(b)
+   '()
+   '((a . 1))))
+
+!(:assert-eq
+  '(+ 1 b)
+  (compile-closure
+   '(+ a b)
+   '(b)
+   '()
+   '((a . 1) (b . 2))))
+
+!(:assert-eq
+  '(lambda (a) (+ a b))
+  (compile-closure
+   '(lambda (a) (+ a b))
+   '(b)
+   '()
+   '((a . 1))))
+
+!(:assert-eq
+  '(let ((a (+ 1 b))) (+ a b))
+  (compile-closure
+   '(let ((a (+ c b))) (+ a b))
+   '(b)
+   '()
+   '((c . 1))))
+
+!(:assert-eq
+  '(+ 1 b)
+  (compile-closure
+   '(+ a b)
+   '(b)
+   '() ;;(current-ram) TODO: RAM is not an exp
+   '((a . 1))))
+
+!(:assert-eq
+  '(+ 1 b)
+  (let ((a 1))
+    (compile-closure
+     '(+ a b)
+     '(b)
+     '()
+     (current-env))))
+
+!(:assert-eq
+  '(+ (f 1) b)
+  (let ((a 1)
+        ;;(f (lambda (x) x))
+        )
+    (compile-closure
+     '(+ (f a) b)
+     '(b)
+     '()
+     (current-env))))
+
+!(:assert-eq
+  '(+ (f 1) b)
+  (letrec ((a 1)
+           ;;(f (lambda (x) x))
+           )
+    (compile-closure
+     '(+ (f a) b)
+     '(b)
+     '()
+     (current-env))))
+
+!(:assert-eq
+  1
+  ((lambda (x) x) 1))
+
+!(:assert-eq
+  1
+  ((compile (lambda (x) x)) 1))
+
+!(:assert-eq
+  12
+  (let ((a 1)
+        (f (lambda (b) (+ a b))))
+    (* (f 2) (f 3))))
+
+!(:assert-eq
+  12
+  (let ((a 1)
+        (f (lambda (b) (+ a b)))
+        (f (compile f)))
```

Comment:
Just a mental note that presumably this would also work if you just had `(f (compile (lambda (b) (+ a b))))`. If I'm missing something, please let me know. I'm not saying that's preferable here — just want to make sure my mental model is accurate.

Reply:
Yes.